src/readinglistbar.cpp

```diff
@@ -65,34 +66,54 @@ void ReadingListBar::setupList()
     }
 }
 
-void ReadingListBar::on_itemActivated(QListWidgetItem* item) // Handles double click and enter key
+void ReadingListBar::onItemActivated(QListWidgetItem* item, Qt::MouseButtons buttons) // Handles double click and enter key
 {   
-    lastDoubleClickTime = QDateTime::currentDateTime();
-    openUrl(item, true);
+    if (buttons & Qt::LeftButton) { itemDoubleClicked(item); }
+    else { itemEntered(item); }
 }
```

Comment:
[`QListWidget::itemActivated()`](https://doc.qt.io/qt-5/qlistwidget.html#itemActivated) is a somewhat confusing signal. Its description reads:

> This signal is emitted when the `item` is activated. The `item` is activated when the user **clicks or double clicks on it, depending on the system configuration**.

Therefore calling `itemDoubleClicked()` from `onItemActiavated()` may be wrong. But do we have to handle any clicks via `onItemActivated()` at all? Why not ignore any item activation events trigerred by mouse clicks and instead take advantage of [`QListWidget::itemClicked()`](https://doc.qt.io/qt-5/qlistwidget.html#itemClicked) and [`QListWidget::itemDoubleClicked`](https://doc.qt.io/qt-5/qlistwidget.html#itemDoubleClicked) signals? I think that `ReadingListBar::onItemPressed()` should be renamed "back to" `ReadingListBar::onItemClicked()` and connected to the ``QListWidget::itemClicked` signal.

Reply:
> It is also activated when the user presses the activation key 

`QListWidget::itemActivated()` is there so as to identify those clicks, but also actions like the `Enter` key for opening an element. 

Using `QListWidget::itemClicked()` and `QListWidget::itemDoubleClicked()` was my first thought as well. After seeing that `QListWidget::itemActivated()` was already there tough, I thought it would be better to just add the `QListWidget::itemPressed` signal so as to have the minimum amount of handlers possible. Is this an issue in your opinion?

Reply:
Amount of handlers is not a metric that we should care about. Software engineers should first of all care about the comprehensibility and maintainability of code. If in a certain situation that translates to the amount of handlers let so be it. But we should not forget how true quality of code should be measured.

Reply:
Got it. Lets continue based on this.

- `onItemActivated` IMO has to stay to keep the action of the **activation** key, but we are now going to return when it is triggered by a click. 
- **Double clicks** will be triggered with the `QListWidget::itemDoubleClicked` and handled by `ReadingListBar::onItemDoubleClicked`.
- `ReadingListBar::onItemPressed()` needs also to be kept from my POV since it is our way for determining **middle clicks**. We can return if we detect left click.
- If the above is true the current behavior will be taken from `ReadingListBar::onItemPressed()` to use for the `ReadingListBar::onItemClicked()` and connect it to the `QListWidget::itemClicked` signal, for the **single left clicks**.

Reply:
> * `ReadingListBar::onItemPressed()` needs also to be kept from my POV since it is our way for determining **middle clicks**. We can return if we detect left click.

Why can't you detect middle clicks inside `ReadingListBar::onItemClicked()`?

Reply:
`QListWidget::itemClicked` does not trigger from middle clicks

Reply:
The description of [`QListWidget::itemClicked`](https://doc.qt.io/qt-5/qlistwidget.html#itemClicked) differs from that of [`QListWidget::itemPressed`](https://doc.qt.io/qt-5/qlistwidget.html#itemPressed) in exactly one word (clicked → pressed). So if your claim is true, then the documentation is defective.
Whatever, please go on with your implementation and we will continue from there.